src/index.js

```diff
@@ -11,26 +11,18 @@ import { assign } from './util';
  *    store.setState({ c: 'd' });   // logs { a: 'b', c: 'd' }
  */
 export default function createStore(state) {
-	let listeners = [];
+	const listeners = [];
 	state = state || {};
 
 	function unsubscribe(listener) {
-		let out = [];
-		for (let i=0; i<listeners.length; i++) {
-			if (listeners[i]===listener) {
-				listener = null;
-			}
-			else {
-				out.push(listeners[i]);
-			}
-		}
-		listeners = out;
+		const i = listeners.indexOf(listener);
+		if (i > -1) listeners.splice(i, 1);
 	}
 
 	function setState(update, overwrite, action) {
-		state = overwrite ? update : assign(assign({}, state), update);
-		let currentListeners = listeners;
```

Comment:
Remove this means that a subscribed function that unsubcribes itself will cause the next listener to be skipped.

Reply:
Are you attempting to hedge the case where a listener cb unsubscribes itself, thus modifies the listeners array? I can see your point now. I may write a test case to prove your example.

Reply:
yup.

Reply:
Does the new test satisfy your concern now?